video/out/vo_gpu_next.c

```diff
@@ -611,6 +611,12 @@ static bool map_frame(pl_gpu gpu, pl_tex *tex, const struct pl_source_frame *src
         .user_data = mpi,
     };
 
+#if PL_API_VER >= 344
+    const struct gl_video_opts *opts = p->opts_cache->opts;
+    if (!opts->alpha_mode)
+        frame->repr.alpha = PL_ALPHA_NONE;
+#endif
```

Comment:
Doing the assignment here will behave strangely when toggling this option at runtime, as `map_frame` is only called once per frame, not on every redraw.

Perhaps the best place for something like this would be inside the `acquire()` callback, although it doesn't feel quite right either. I'm not sure how to handle this whole situation elegantly - it seems that updating the properties of multiple frames dynamically is a fundamentally difficult problem with the current API.

Maybe `pl_render_params` should take something like a frame metadata override.

Reply:
Oh yeah it doesn't even work on runtime.

Reply:
The alpha option is no more.